sympy/physics/optics/gaussopt.py

```diff
@@ -526,18 +527,18 @@ class BeamParameter(Expr):
     # subclass it. See:
     # https://groups.google.com/d/topic/sympy/7XkU07NRBEs/discussion
 
-    def __new__(cls, wavelen, z, z_r=None, w=None):
+    def __new__(cls, wavelen, z, n=1, z_r=None, w=None):
```

Comment:
Why is n=1 here by default?

Reply:
Because initially we assume beam waist is being calculated in vacuum having a refractive index of 1 ,unless medium has been changed .

Reply:
You should sympify the extra argument that you have added so that it will be converted into instances of sympy.Integer instead of being treated as python `int`. Then try adding the extra argument into test args function and it should work.

Reply:
Thanks for the suggestion ,now I understood why there was an error in the first place !